runtime/core/decoder/ov_asr_model.cc

```diff
@@ -0,0 +1,490 @@
+  }
+
+  // Reset cnn_cache
+  cnn_cache_.resize(
+      num_blocks_ * encoder_output_size_ * (cnn_module_kernel_ - 1), 0.0);
+  ov::Shape cnn_cache_shape = {num_blocks_, 1, encoder_output_size_,
+                               cnn_module_kernel_ - 1};
+  cnn_cache_ov_ =
+      ov::Tensor(ov::element::f32, cnn_cache_shape, cnn_cache_.data());
+}
+
+void OVAsrModel::ForwardEncoderFunc(
+    const std::vector<std::vector<float>>& chunk_feats,
+    std::vector<std::vector<float>>* out_prob) {
+  // 1. Prepare OV required data, splice cached_feature_ and chunk_feats
+  // chunk
+  int num_frames = cached_feature_.size() + chunk_feats.size();
+  const int feature_dim = chunk_feats[0].size();
+  std::vector<float> feats;
+  for (size_t i = 0; i < cached_feature_.size(); ++i) {
+    for (size_t j = 0; j < feature_dim; ++j) {
+      feats.emplace_back(cached_feature_[i][j]);
+    }
+  }
+  for (size_t i = 0; i < chunk_feats.size(); ++i) {
+    for (size_t j = 0; j < feature_dim; ++j) {
+      feats.emplace_back(chunk_feats[i][j]);
+    }
+  }
+
+  ov::Shape feats_shape = {1, num_frames, feature_dim};
+  ov::Tensor feats_ov = ov::Tensor(ov::element::f32, feats_shape, feats.data());
+
+  // offset
+  int64_t offset_int64 = static_cast<int64_t>(offset_);
+  ov::Tensor offset_ov =
+      ov::Tensor(ov::element::i64, ov::Shape(), &offset_int64);
+
+  // required_cache_size
+  int64_t required_cache_size = chunk_size_ * num_left_chunks_;
+  ov::Tensor required_cache_size_ov =
+      ov::Tensor(ov::element::i64, ov::Shape(), &required_cache_size);
+
+  // att_mask
+  ov::Tensor att_mask_ov;
+  std::vector<uint8_t> att_mask(required_cache_size + chunk_size_, 1);
+  if (num_left_chunks_ > 0) {
+    int chunk_idx = offset_ / chunk_size_ - num_left_chunks_;
+    if (chunk_idx < num_left_chunks_) {
+      for (int i = 0; i < (num_left_chunks_ - chunk_idx) * chunk_size_; ++i) {
+        att_mask[i] = 0;
+      }
+    }
+    ov::Shape att_mask_shape = {1, 1, required_cache_size + chunk_size_};
+    att_mask_ov = ov::Tensor(ov::element::boolean, att_mask_shape,
+                             reinterpret_cast<bool*>(att_mask.data()));
+  }
+
+  // set input tensor
+  size_t idx = 0;
+
+  for (auto name : encoder_input_names_) {
+    if (name == "chunk") {
+      encoder_infer_->set_tensor(encoder_inputs_map_[name], feats_ov);
+    } else if (name == "offset") {
+      encoder_infer_->set_tensor(encoder_inputs_map_[name], offset_ov);
+    } else if (name == "required_cache_size") {
+      encoder_infer_->set_tensor(encoder_inputs_map_[name],
+                                 required_cache_size_ov);
+    } else if (name == "att_cache") {
+      encoder_infer_->set_tensor(encoder_inputs_map_[name], att_cache_ov_);
+    } else if (name == "cnn_cache") {
+      encoder_infer_->set_tensor(encoder_inputs_map_[name], cnn_cache_ov_);
+    } else if (name == "att_mask") {
+      encoder_infer_->set_tensor(encoder_inputs_map_[name], att_mask_ov);
+    }
+  }
```

Comment:
seems that `encoder_input_names_` is only used in this for-loop, I think we can also for-loop `encoder_inputs_map_` to get names thus there is no need to maintain extra variable (`encoder_input_names_`)